### PR TITLE
YD-252 Activity pagination

### PR DESCRIPTION
**Pull request by aristotelos**
Using oldest goal creation time to determine total number of days/weeks available
Default page size 3 for days, 2 for weeks
Removed duplicate old style tests
Added some asserts for pagination in tests

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -148,9 +148,13 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
 		analysisService.postToAnalysisEngine(user, categories, url, YonaServer.relativeDateTimeStringToZonedDateTime(relativeDateTimeString))
 	}
 
-	void assertWeekOverviewBasics(response, numberOfReportedGoals)
+	void assertWeekOverviewBasics(response, numberOfReportedGoals, expectedTotalElements)
 	{
+		def expectedPageSize = 2
```
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: The next method makes this a default parameter. Can't you apply that elegant solution here too?

File: appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy [hunk 2]
```diff
@@ -42,17 +67,37 @@ class ActivityTest extends AbstractAppServiceIntegrationTest
...
-			"Wed" : [[goalUrl:budgetGoalNewsUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: []]], [goalUrl:timeZoneGoalSocialUrl, data: [goalAccomplished: false, minutesBeyondGoal: 1, spread: [68 : 1]]]],
-			"Thu" : [[goalUrl:budgetGoalNewsUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: []]], [goalUrl:timeZoneGoalSocialUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: [48 : 1]]]],
-			"Fri" : [[goalUrl:budgetGoalNewsUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: []]], [goalUrl:timeZoneGoalSocialUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: [ : ]]]],
-			"Sat" : [[goalUrl:budgetGoalNewsUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: []]], [goalUrl:timeZoneGoalSocialUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: [ : ]]]]]
+			"Wed" : [
+				[goalUrl:budgetGoalNewsUrl, data: [goalAccomplished: true, minutesBeyondGoal: 0, spread: []]],
+				[goalUrl:timeZoneGoalSocialUrl, data: [goalAccomplished: false, minutesBeyondGoal: 1, spread: [68 : 1]]]
+			],
```
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy]: I like this formatting, but for me, the code formatter joins the lines every time.

File: core/src/main/java/nu/yona/server/analysis/service/ActivityService.java [hunk 3]
```diff
@@ -83,7 +84,26 @@
 		return new PageImpl<WeekActivityOverviewDTO>(
 				weekActivityEntitiesByZonedDate.entrySet().stream().sorted((e1, e2) -> e2.getKey().compareTo(e1.getKey()))
 						.map(e -> WeekActivityOverviewDTO.createInstance(e.getKey(), e.getValue())).collect(Collectors.toList()),
-				pageable, yonaProperties.getAnalysisService().getWeeksActivityMemory());
+				pageable, getTotal(userAnonymized, ChronoUnit.WEEKS));
+	}
+
+	private long getTotal(UserAnonymizedDTO userAnonymized, ChronoUnit timeUnit)
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/analysis/service/ActivityService.java]: Please rename into ``getTotalPageableItems``.

**Comment by Bert-R** [comment]
Looks good. Added a few small comments.

File: appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy (hunk 2 above)
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy]: For me it splits those lines. Is the "Length after which lists are 'long' (long lists are wrapped)" for you equal to 30?

**Comment by aristotelos** [comment]
Processed review comments

File: appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy (hunk 2 above)
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy]: Yes, that's the same. Weird.

**Comment by Bert-R** [comment]
Looks good. Once the current build is done, I'll merge this one.

**Pull request merged**